src/devices/CharacterLcd/LcdConsole.cs

```diff
@@ -0,0 +1,537 @@
+        {
+            _lcd = lcd;
+            _shouldDispose = shouldDispose;
+            _romType = romType;
+            _characterEncoding = null;
+            Size = lcd.Size;
+            _currentData = new StringBuilder[Size.Height];
+            CursorLeft = 0;
+            CursorTop = 0;
+            NewLineSleepTime = TimeSpan.Zero;
+            _lock = new object();
+            _lcd.UnderlineCursorVisible = false;
+            _lcd.BlinkingCursorVisible = false;
+            _lcd.DisplayOn = true;
+            _lcd.BacklightOn = true;
+            ClearStringBuffer();
+            _lcd.Clear();
+        }
+
+        /// <summary>
+        /// Position of the cursor, from left.
+        /// Note: May be outside the bounds of the display.
+        /// </summary>
+        public int CursorLeft
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Position of the cursor, from top
+        /// Note: May be outside the bounds of the display.
+        /// </summary>
+        public int CursorTop
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// If larger than Zero, an extra wait is introduced each time a newline is written. Can be used to render entire paragraphs to the display, 
+        /// but keeping it readable. 
+        /// </summary>
+        public TimeSpan NewLineSleepTime
+        {
+            get
+            {
+                return _newLineSleepTime;
+            }
+
+            set
+            {
+                if (value < TimeSpan.Zero)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), "Timespan must be positive");
+                }
+                _newLineSleepTime = value;
+            }
+        }
+
+        /// <summary>
+        /// Enables or disables the backlight
+        /// </summary>
+        public bool BacklightOn
+        {
+            get
+            {
+                lock (_lock)
+                {
+                    return _lcd.BacklightOn;
+                }
+            }
+            set
+            {
+                lock (_lock)
+                {
+                    _lcd.BacklightOn = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Enables or disables the display
+        /// </summary>
+        public bool DisplayOn
+        {
+            get
+            {
+                lock (_lock)
+                {
+                    return _lcd.DisplayOn;
+                }
+            }
+            set
+            {
+                lock (_lock)
+                {
+                    _lcd.DisplayOn = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sets the Line Feed Mode. 
+        /// This defines what happens when writting past the end of the line/screen.
+        /// </summary>
+        public LineFeedMode LineFeedMode
```

Comment:
I'd call this and enum `WordWrap` or `WordWrapOption`  (ing suffix fine, whichever sounds better) and move ScrollUp as a separate property

Reply:
Called it LineWrapMode now.